### PR TITLE
feat: Map Ironic provisioning status to higher level Nautobot Status

### PR DESCRIPTION
**Pull request by skrobul**
Closes PUC-408

Goes together with https://github.com/RSS-Engineering/undercloud-nautobot/pull/187

### CODE REVIEW

File: argo-workflows/ironic-to-nautobot-sync/workflowtemplates/sync.yaml [hunk 1]
```diff
@@ -12,13 +12,11 @@ spec:
       container:
         image: ghcr.io/rackerlabs/understack/ironic-nautobot-client:latest
         command:
-          - nautobot-update-cf
+          - synchronize-provision-state-to-nautobot
```
Comment (cardoe) [inline argo-workflows/ironic-to-nautobot-sync/workflowtemplates/sync.yaml:15]: here it's `synchronize-provision-state-to-nautobot` but....

File: python/understack-workflows/pyproject.toml [hunk 2]
```diff
@@ -44,6 +44,7 @@ pytest-mock = "^3.14.0"
 synchronize-interfaces = "understack_workflows.main.synchronize_interfaces:main"
 synchronize-obm-creds = "understack_workflows.main.synchronize_obm_creds:main"
 synchronize-server = "understack_workflows.main.synchronize_server:main"
+synchronize-provision-state = "understack_workflows.main.synchronize_provision_state:main"
```
Comment (cardoe) [inline python/understack-workflows/pyproject.toml:47]: here it's `synchronize-provision-state`

**Review by cardoe — COMMENTED** [review]
Just a small disconnect in the workflow YAML from the executable name. It does make me wonder if we should call the binary `synchronize-ironic-to-nautobot` or something like that. Since you have the `--provision-state` as a flag. Maybe in the future there's other bits we want to push and we can add more flags?

File: argo-workflows/ironic-to-nautobot-sync/workflowtemplates/sync.yaml (hunk 1 above)
Comment (skrobul) [reply argo-workflows/ironic-to-nautobot-sync/workflowtemplates/sync.yaml:15]: good catch, it's supposed to be `synchronize-provision-state`, will fix.

**Review by cardoe — APPROVED** [review]

**Pull request merged**